internal/receiver/discoveryreceiver/config.go

```diff
@@ -103,7 +103,7 @@ func (cfg *Config) Validate() error {
 		}
 	}
 
-	if cfg.WatchObservers == nil || len(cfg.WatchObservers) == 0 {
+	if cfg.WatchObservers == nil {
```

Comment:
```suggestion
	if len(cfg.WatchObservers) == 0 {
```

From my understanding, I don't _think_ it matters here which condition we take, but I'd slightly favor the length check more than nil, which is also what has been [done upstream](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/34724/files). I'd be interested to know though if there's a reason/preference to do the `nil` check instead.